### DISCUSSION TITLE
infrastructure/openwrt/disable-ipv6/

### DISCUSSION THREAD

**Discussion by giscus** [fire1ce/3os.org#64] [General]
# infrastructure/openwrt/disable-ipv6/

How to disable IPV6 on your OpenWrt router

https://3os.org/infrastructure/openwrt/disable-ipv6/

**Comment by allenwalker3** [comment]
Is there a reason you are against IPv6?

**Reply by fire1ce** [reply]
It's not that I'm against IPv6. 
IPv6 meant to solve one big problem - the limit of valid IPv4 address in the world. 
There is no valid reason to use IPv6 Protocol at your home network - the RFC1918 has you covered. 
When using IPv6 you add unnecessary complexity to your network - specialy when you what to monitor it.
The only reason you need to use IPv6 protocol for comunication is when your ISP will provide you only IPv6 address.

**Reply by allenwalker3** [reply]
Doesn't IPv6 offer other things other than unlimited ip addresses? Ipv6 is more efficient than Ipv4 from what I understand.

**Reply by jaimet** [reply]
> Ipv6 is more efficient than Ipv4 from what I understand.

"Efficient" in what sense?

**Reply by allenwalker3** [reply]
At a very highlevel: 

IPv6 simplifies the router’s task compared to IPv4.
IPv6 is more compatible to mobile networks than IPv4.
IPv6 allows for bigger payloads than what is allowed in IPv4.

**Comment by jaimet** [comment]
Hi! Why do you disable prefix delegation on the lan interface if it's now an "IPv4-only" interface? (Your very first command disables IPv6 on this interface). I thought that prefix delegation was an IPv6-only concept... :thinking:

**Comment by TheBlueJester** [comment]
Small Remark in defence of disabling IPV6:
IPv6 can make blocking ads more difficult / complicated. 
So disabling it to improve my ability to block Ads is of greater benefit than the benefits of IPv6 in my case. 
You can google "IPv6 and Adblocking" to see that others share this issue. 

I know this is an old "post" but I've just come across this via the webpage at 3os.org.
I thought I might add my reason for wanting IPv6 disabled, both for people who see this page in the future, or if people want to give me constructive critisim on my approach.

**Reply by allenwalker3** [reply]
Agreed with DNS blocking. I run an ad blocker locally but I'm never 100% sure it's using my IPv6 dns or some public one. Just looking at the addresses makes it almost impossible to determine.

**Comment by 0xorial** [comment]
is it safe to delete wan6 interface?

**Reply by allenwalker3** [reply]
Probably. Many ISPs don't even support it anyway. And if they do it's not required. IPv4 will never die and can be used by itself without any risk of internet access disruption. Ipv6 is basically being used in Ipv6 -> ipv4 tunneling, so that any device can get a public Ipv6 address that can't get an available ipv4 public address

**Comment by CyberKillSec** [comment]
Hello all,
Thank you very much fire1ce, this is exactly what I've been looking for for months.
Looking forward to reading you cordially.

**Comment by nwgat** [comment]
if your network is unstable, you might want to try to disable ipv6, ive had issues with something on my network not playing nice with ipv6 and i don't know what it is

**Comment by d0zingcat** [comment]
> When using IPv6 you add unnecessary complexity to your network - specialy when you what to monitor it.

Many thanks; this post saved my day. As mentioned earlier, IPv6 introduces complexity to my network. I've encountered various issues, including slow network speeds and DNS resolution failures. While these problems may have resulted from my incorrect configurations, it's undeniably more challenging for me to grasp all the intricacies of IPv6 configurations when compared to IPv4.

**Comment by Vitus-TL** [comment]
Hi, still got some link-local addresses left, which I don't know how to remove/disable:
root@iotrouter:~# ip -6 a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 state UNKNOWN qlen 1000
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
7: br-lan: <BROADCAST,UP,LOWER_UP> mtu 1500 state UP qlen 1000
    inet6 fe80::XXXXX/64 scope link 
       valid_lft forever preferred_lft forever
8: br-lan.254: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 state UP qlen 1000
    inet6 fe80::XXXX/64 scope link 
       valid_lft forever preferred_lft forever

**Comment by jaimet** [comment]
> Hi, still got some link-local addresses left, which I don't know how to remove/disable

Re the loopback device, I have heard that bad things can happen if you remove the ipv6 interface and address, hence I leave it there (I haven't done any research to find out if this is true). If I wanted to get rid of it, I'd try an `option ipv6 '0'` line in the relevant stanza.

Re the `br-lan` devices, do you already have `option ipv6 '0'` lines in those stanzas too?

**Comment by e-e-eric** [comment]
Thanks! Solved this bugbear of mine.

openwrt isn't my DHCP server so the ipv6 addresses were causing unnecessary complications, particularly with pxeboot.
